src/reverse_route.coffee

```diff
@@ -1,11 +1,22 @@
-class ReverseRoute extends String
+class ReverseRoute
 
   constructor: ({@path, @method}) ->
-    super @path
+    @length = @path.length
 
+    # Enable single character access
+    @[i] = character for character, i in @path
 
-  toString: ->
-    @path
+
+  # Forward String methods
+  indexOf: -> @path.indexOf arguments...
+
```

Comment:
Why do we need all of these methods delegated? I'd say keep it simple for now and just have the toString / valueOf methods.